.github/workflows/rust.yml

```diff
@@ -0,0 +1,48 @@
+name: Rust test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        rust: [stable, nightly, beta]
+
+    steps:
+      - uses: hecrj/setup-rust-action@v1
+        with:
+          rust-version: ${{ matrix.rust }}
+      - uses: actions/checkout@master
+      - name: Build
+        run: cargo build --examples --lib --bins
+      - name: Test
+        run: cargo test --verbose --examples --lib --bins
```

Comment:
why do we need to separate between `cargo build` and `cargo test`?

Reply:
Its to not get the super verbpse build output that comes from implicit build from running cargo test with verbose flag.

Reply:
ok got it